src/__tests__/field/email.test.js

```diff
@@ -0,0 +1,22 @@
+import Immutable from 'immutable';
+
+describe('field/email', () => {
+  let email;
+  beforeEach(() => {
+    jest.resetModules();
+
+    jest.mock('field/index', () => ({
+      setField: jest.fn()
+    }));
+
+    email = require('field/email');
+  });
+  describe('setEmail()', () => {
+    it(`removes spaces`, () => {
+      email.setEmail(Immutable.fromJS({}), ' email@te  st.com ');
```

Comment:
While I agree leading and trailing whitespaces should be trimmed, i think having whitespaces on the middle of an email address or username is an "invalid value" rather.

Reply:
why show "invalid field" if we can remove them at once and move on?

Reply:
Was just typing this out, actually. Fully agree (if spaces are actually invalid characters ... are they excluded by the platform?)

Reply:
Because it's a user's typing error and we should point them that it's an invalid value and they should fix it. If we don't, they might think they registered using "this email @address.com" and when they try to log in on other platform they won't be allowed.

Also, A0 username regex: `^[a-zA-Z0-9_]+$`

Screenshot below from gmail when typing an email address:
![image](https://user-images.githubusercontent.com/3900123/37173919-c380b92a-22f3-11e8-9819-6bd51c097352.png)

Reply:
fixed